src/test-helpers.tsx

```diff
@@ -143,6 +144,7 @@ export const fakeVersion: ExternalVersionWithContent = Object.freeze({
 export const createExternalVersionWithEntries = (
   partialEntries: ({ path: string } & Partial<ExternalVersionEntry>)[],
   versionProps: Partial<ExternalVersionWithContent> = {},
+  fileProps: Partial<ExternalVersionFileWithDiff> = {},
```

Comment:
Adding an extra positional argument makes these functions calls hard to read (and they could be hard to maintain). 

Since `partialEntries` is not optional but `versionProps` is, I suggest switching to a signature where the second positional argument contains all optional parameters needed by existing tests (this keeps it simple). I reviewed the current calls and I think these are the only optional parameters needed:

```diff
diff --git a/src/test-helpers.tsx b/src/test-helpers.tsx
index 2a7e67a..e321af3 100644
--- a/src/test-helpers.tsx
+++ b/src/test-helpers.tsx
@@ -143,12 +143,19 @@ export const fakeVersion: ExternalVersionWithContent = Object.freeze({
 
 export const createExternalVersionWithEntries = (
   partialEntries: ({ path: string } & Partial<ExternalVersionEntry>)[],
-  versionProps: Partial<ExternalVersionWithContent> = {},
-  fileProps: Partial<ExternalVersionFileWithDiff> = {},
+  {
+    fileDiff,
+    selected_file = fakeVersionFile.selected_file,
+    id = fakeVersion.id,
+  }: {
+    fileDiff?: typeof fakeExternalDiff;
+    selected_file?: typeof fakeVersionFile.selected_file;
+    id?: typeof fakeVersion.id;
+  } = {},
 ) => {
   return {
     ...fakeVersion,
-    ...versionProps,
+    id,
     file: {
       ...fakeVersion.file,
       entries: partialEntries.reduce((entries, file) => {
@@ -162,7 +169,8 @@ export const createExternalVersionWithEntries = (
           ...entries,
         };
       }, {}),
-      ...fileProps,
+      diff: fileDiff,
+      selected_file,
     },
   };
 };
```

Maybe that will work? It might need more parameters.